ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml

```diff
@@ -260,16 +275,20 @@ QObject {
 
             if (password !== "") {
                 var actionResult = ""
-                if (request.method === SessionRequest.methods.personalSign.name) {
-                    // TODO #14756: clarify why prefixing the message fails the test app https://react-app.walletconnect.com/
-                    //let finalMessage = "\x19Ethereum Signed Message:\n" + originalMessage.length + originalMessage
+                if (request.method === SessionRequest.methods.personalSign.name ||
+                    request.method === SessionRequest.methods.sign.name)
+                {
                     actionResult = store.signMessage(request.topic, request.id,
```

Comment:
not sure if we should add the prefix to `eth_sign` as well, as it's not mentioned in EIP191. I think we can remove it, [as Metamask soon will](https://github.com/MetaMask/metamask-improvement-proposals/blob/main/MIPs/mip-3.md), but for now I guess it's fine to keep it as is and will discuss in the wallet weekly whether we should disable it.

Reply:
I was actually following Wallet Connect recommendations here https://docs.walletconnect.com/advanced/multichain/rpc-reference/ethereum-rpc#eth_sign
Those propose to have the legacy also prefixed to protect against attacks. I suppose this might break some really old code but keep safer newer code. I was thinking to redirect `eth_sign` to the signing only process when the testing encounters an issue with the prefixed message signature in the case of a known dApp

Reply:
I restored the legacy implementation for the `eth_sign` with the latest changes